pkg/brokerapi/openservicebroker/open_service_broker_client.go

```diff
@@ -78,59 +116,78 @@ func (c *openServiceBrokerClient) GetCatalog() (*brokerapi.Catalog, error) {
 	return &catalog, nil
 }
 
-func (c *openServiceBrokerClient) CreateServiceInstance(ID string, req *brokerapi.ServiceInstanceRequest) (*brokerapi.ServiceInstance, error) {
-	jsonBytes, err := json.Marshal(req)
-	if err != nil {
-		return nil, err
-	}
-
+func (c *openServiceBrokerClient) CreateServiceInstance(ID string, req *brokerapi.CreateServiceInstanceRequest) (*brokerapi.CreateServiceInstanceResponse, error) {
 	url := fmt.Sprintf(serviceInstanceFormatString, c.broker.Spec.URL, ID)
-
 	// TODO: Handle the auth
-	createHTTPReq, err := http.NewRequest("PUT", url, bytes.NewReader(jsonBytes))
+	resp, err := util.SendRequest(c.client, http.MethodPut, url, req)
 	if err != nil {
-		return nil, err
+		return nil, errRequest{message: err.Error()}
 	}
+	defer resp.Body.Close()
 
-	glog.Infof("Doing a request to: %s", url)
-	resp, err := c.client.Do(createHTTPReq)
-	if err != nil {
-		return nil, err
+	createServiceInstanceResponse := brokerapi.CreateServiceInstanceResponse{}
+	if err := util.ResponseBodyToObject(resp, &createServiceInstanceResponse); err != nil {
+		return nil, errResponse{message: err.Error()}
 	}
-	defer resp.Body.Close()
 
-	// TODO: Align this with the actual response model.
-	si := brokerapi.ServiceInstance{}
-	if err := util.ResponseBodyToObject(resp, &si); err != nil {
-		return nil, err
+	switch resp.StatusCode {
+	case http.StatusCreated:
+		return &createServiceInstanceResponse, nil
+	case http.StatusOK:
+		return &createServiceInstanceResponse, nil
+	case http.StatusAccepted:
+		glog.Infof("Asynchronous response received. Polling broker.")
+		if err := c.pollBroker(ID, createServiceInstanceResponse.Operation); err != nil {
```

Comment:
Yeah, that makes sense. It seems me like if we're going to refactor for more dynamic feedback, though, it might make sense to move the polling loop to the controller. This would allow for us to keep track of not only the fact that the instance is asynchronously provisioning, but also have an easy way to store updated information about the provisioning process that might be passed via the `description` attribute of the last operation response body (documented [here](https://docs.cloudfoundry.org/services/api.html#polling)).

In that case, the broker client would pass back a `Created`, `Provisioning`, or `Failed` response upon trying to provision, and the controller would then be responsible to loop and call a function in the broker client that would pass back the current state/description of the operation.

Would that approach be appropriate?